### ISSUE TITLE
Performance issue - loading images multiple times in page

### ISSUE THREAD

**Issue by lackzinsky** [720kb/angular-socialshare#67]
Hi 720Kb,

First of all thanks, the angular-socialshare module is great. Extremely easy to implement many networks at once and provides a critical functionality that many developers need. I just seem to have a performance issue that I cannot find a solution for.

I have a page with 500 images (10MB) that I display with a ng-repeat. I am also adding one Facebook feed share button to each image with several attributes (text,caption,etc) including a socialshare-media="{{getImgUrl(photo)}}". The getImgUrl is a function that returns the url to the given image.

When I click on the Facebook share button the dialog shows up containing all the information I'm passing through the attributes, including the images. The problem is that checking the network console in my browser it seems all 500 images are re-downloaded in the page when I click the share button. This is hurting performance enormously. It seems the issue occurs in Safari on my mac though it doesn't seem like it happen in Chrome or Firefox.

Any ideas? I can attach some code and provide more info if needed

Regards,
Diego

**Comment by 45kb** [comment]
Hi @lackzinsky  how do you retrieve the images? Could you please be so gentle to show some of the code, specially the getImgUrl.

Thank you

**Comment by lackzinsky** [comment]
Thanks for your quick response.
I think I made a mistake though. I've just isolated the code with the list of images and Facebook share buttons, and it seems the double loading of images does not happen at all now. The issue must be with some other part of the code I'm using. The reason I thought it could be due to the socialshare module is because it seems the re-loading of images triggers when I click the share button. I will keep on investigating and let you know what the outcome is.

Diego
